urlb3290c8e9e_craigonic_jasl_tgz-pJ8-DriverJ8/src/Driver.java

```diff
@@ -0,0 +1,555 @@
+                                       6,2,8,8,false,12,4,false,
+                                       Classification.Classifications.ELITE,
+                                       true,true,0);
+
+        russianSquad.setIdentity("A");
+
+        // Display all of the entered values for this instance using the
+        // toText() method.
+
+        System.out.println("\nSquad.toText() output:\n\n" +
+                           russianSquad.toText());
+
+        // Display an abbreviated description of this instance using the
+        // toString() method.
+
+        System.out.println("Squad.toString() output:\n\n" +
+                           russianSquad.toString());
+
+        // Display all of the entered values for this instance using the
+        // toJSON() method.
+
+        System.out.println("\nSquad.toJSON() output:\n\n" +
+                           russianSquad.toJSON());
+
+        // Create an array of Unit objects. These will be used to reference a
+        // Leader instance and several Squad instances. These class types are
+        // derived from Unit.
+
+        System.out.println("\nBuilding Unit array with a Leader & 3 Squads\n");
+
+        Unit[] UnitList = new Unit[4];
+
+        UnitList[0] = new Leader(Nationality.Nationalities.AMERICAN,
+                                 UnitType.InfantryTypes.NONE,9,9,4,-1);
+
+        ((Leader)UnitList[0]).setIdentity("Sgt. Slaughter");
+
+        UnitList[1] = new Squad(Nationality.Nationalities.AMERICAN,
+                                UnitType.InfantryTypes.NONE,
+                                6,6,6,6,false,11,4,false,
+                                Classification.Classifications.FIRST_LINE,
+                                true,false,3);
+
+        ((Squad)UnitList[1]).setIdentity("X");
+
+        UnitList[2] = new Squad(Nationality.Nationalities.AMERICAN,
+                                UnitType.InfantryTypes.NONE,
+                                6,6,6,6,false,11,4,false,
+                                Classification.Classifications.FIRST_LINE,
+                                true,false,3);
+
+        ((Squad)UnitList[2]).setIdentity("Y");
+
+        UnitList[3] = new Squad(Nationality.Nationalities.AMERICAN,
+                                UnitType.InfantryTypes.NONE,
+                                6,6,6,6,false,11,4,false,
+                                Classification.Classifications.FIRST_LINE,
+                                true,false,3);
+
+        ((Squad)UnitList[3]).setIdentity("Z");
+
+        System.out.println("Displaying Unit array with a Leader & 3 Squads");
+
+        Leader leaderObject = null;
+        Squad  squadObject  = null;
+
+        for (int i = 0; i < 4; i++)
+        {
+            System.out.println("\nUnitList[" + i + "]:\t" +
+                               UnitList[i].toString());
+
+            if (Description.Descriptions.LEADER.toString() == UnitList[i].description())
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.